LICENSE

```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
```

Comment:
I'd prefer [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0.txt), and if updating the license, could you send that as its own PR instead of having it buried in with other changes?

Reply:
Ok

Reply:
Ok

Reply:
Ok

Reply:
Ok

Reply:
I will start it after this PR is closed.

Reply:
I added a license file in master. Please pull and rebase and you should be able to drop this file and some of your other miscellaneous changes.